src/test/server/ServerStatus_test.cpp

```diff
@@ -422,117 +516,346 @@ class ServerStatus_test :
     }
 
     void
-    testAdminRequest(std::string const& proto, bool admin, bool credentials)
+    testAuth(boost::asio::yield_context& yield)
     {
-        testcase << "Admin request over " << proto <<
-            ", config " << (admin ? "enabled" : "disabled") <<
-            ", credentials " << (credentials ? "" : "not ") << "set";
-        using namespace jtx;
-        Env env {*this, makeConfig(proto, admin, credentials)};
+        testcase ("Server with authorization");
 
-        auto const user = env.app().config()
-            [boost::starts_with(proto, "h") ? "port_rpc" : "port_ws"].
-            get<std::string>("admin_user");
+        using namespace test::jtx;
+        Env env {*this, envconfig([](std::unique_ptr<Config> cfg) {
+            (*cfg)["port_rpc"].set("user","me");
+            (*cfg)["port_rpc"].set("password","secret");
+            return cfg;
+        })};
 
-        auto const password = env.app().config()
-            [boost::starts_with(proto, "h") ? "port_rpc" : "port_ws"].
-            get<std::string>("admin_password");
+        Json::Value jr;
+        jr[jss::method] = "server_info";
+        beast::http::response<beast::http::string_body> resp;
+        boost::system::error_code ec;
+        doHTTPRequest(env, yield, false, resp, ec, to_string(jr));
+        BEAST_EXPECT(resp.status == 403);
+
+        beast::http::fields auth;
+        auth.insert("Authorization", "");
+        doHTTPRequest(env, yield, false, resp, ec, to_string(jr), auth);
+        BEAST_EXPECT(resp.status == 403);
+
+        auth.replace("Authorization", "Basic NOT-VALID");
+        doHTTPRequest(env, yield, false, resp, ec, to_string(jr), auth);
+        BEAST_EXPECT(resp.status == 403);
+
+        auth.replace("Authorization", "Basic " + beast::detail::base64_encode("me:badpass"));
+        doHTTPRequest(env, yield, false, resp, ec, to_string(jr), auth);
+        BEAST_EXPECT(resp.status == 403);
+
+        // finally if we use the right user/pass, we should get a 200
+        auth.replace("Authorization", "Basic " + beast::detail::base64_encode(
+            env.app().config().section("port_rpc").
+                get<std::string>("user").value() + ":" +
+            env.app().config().section("port_rpc").
+                get<std::string>("password").value()
+        ));
+
+        doHTTPRequest(env, yield, false, resp, ec, to_string(jr), auth);
+        BEAST_EXPECT(resp.status == 200);
+        BEAST_EXPECT(! resp.body.empty());
+    }
 
-        Json::Value jrr;
+    void
+    testLimit(boost::asio::yield_context& yield)
+    {
+        testcase ("Server with connection limit");
```

Comment:
it looks like the `limit` param was added in 2016, so I doubt that it is deprecated. I will add it to the config doc.